hubblestack/utils/osquery_lib.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import os
+import salt.modules.cmdmod
+import json
+
+__salt__ = {'cmd.run': salt.modules.cmdmod._run_quiet,
+            'cmd.run_all': salt.modules.cmdmod.run_all}
+
+log = logging.getLogger(__name__)
+
+def query(query_sql='', osquery_path=None, args=None, max_file_size=104857600, timeout=10000):
+  try:
+    if not query_sql:
+      return ''
+    if not osquery_path:
+      if not os.path.isfile(__grains__['osquerybinpath']):
+        log.error('osquery binary not found: %s', __grains__['osquerybinpath'])
+        return ''
+      cmd = [__grains__['osquerybinpath'], '--read_max', max_file_size, '--json', query_sql]
+    else:
+      if not os.path.isfile(osquery_path):
+        log.error('osquery binary not found: %s', osquery_path)
+        return ''
+      cmd = [osquery_path, '--read_max', max_file_size, '--json', query_sql]
+    if isinstance(args, (list, tuple)):
+      cmd.extend(args)
+
+    # Run the command
+
+    res = __salt__['cmd.run_all'](cmd, timeout=timeout, python_shell=False)
+    if res['retcode'] == 0:
+      ret = json.loads(res['stdout'])
+      if len(ret) != 0:
+        for result in ret:
+          for key, value in result.items():
+            if value and isinstance(value, unicode):
+              return value
```

Comment:
For queries returning multiple results. We are returning just first.
Here Example: dockerd, dockerd-current, dockerd-default

This case might fit here. But, this function is for running any query (as the name suggests).
What will happen if caller is expecting all such output.